pkg/innerring/processors/netmap/process_peers.go

```diff
@@ -18,6 +18,19 @@ func (np *Processor) processAddPeer(ev netmapEvent.AddPeer) {
 		return
 	}
 
+	// check if notary transaction is valid, see #976
+	if originalRequest := ev.NotaryRequest(); originalRequest != nil {
+		tx := originalRequest.MainTransaction
+		ok, err := np.netmapClient.Morph().IsValidScript(tx.Script, tx.Signers)
```

Comment:
IIUC, currently `tx.Signers` still include alphabet multisigner. And `addPeer` still checks for this alphabet signer (https://github.com/nspcc-dev/neofs-contract/blob/8553320e1c611d0462339f48fbd5b36358f33b83/netmap/netmap_contract.go#L213). As a result, witness check for new public key will be skipped, isn't it?

We need to refactor `addPeer` code along with these changes.

Reply:
Sure, `addPeer` will be ([most probably](https://github.com/nspcc-dev/neofs-contract/issues/154#issuecomment-970341435)) split into two methods and it will contain mandatory node witness check.
```go
		publicKey := nodeInfo[2:35] // offset:2, len:33
		if !runtime.CheckWitness(publicKey) {
		}
```